src/Shared/Buffers.MemoryPool/MemoryPoolSlab.cs

```diff
@@ -11,70 +9,34 @@ namespace System.Buffers
     /// </summary>
     internal class MemoryPoolSlab : IDisposable
     {
-        /// <summary>
-        /// This handle pins the managed array in memory until the slab is disposed. This prevents it from being
-        /// relocated and enables any subsections of the array to be used as native memory pointers to P/Invoked API calls.
-        /// </summary>
-        private GCHandle _gcHandle;
-        private bool _isDisposed;
-
-        public MemoryPoolSlab(byte[] data)
+        private MemoryPoolSlab(byte[] pinnedData)
         {
-            Array = data;
-            _gcHandle = GCHandle.Alloc(data, GCHandleType.Pinned);
-            NativePointer = _gcHandle.AddrOfPinnedObject();
+            PinnedArray = pinnedData;
         }
 
         /// <summary>
         /// True as long as the blocks from this slab are to be considered returnable to the pool. In order to shrink the
         /// memory pool size an entire slab must be removed. That is done by (1) setting IsActive to false and removing the
         /// slab from the pool's _slabs collection, (2) as each block currently in use is Return()ed to the pool it will
         /// be allowed to be garbage collected rather than re-pooled, and (3) when all block tracking objects are garbage
-        /// collected and the slab is no longer references the slab will be garbage collected and the memory unpinned will
-        /// be unpinned by the slab's Dispose.
+        /// collected and the slab is no longer references the slab will be garbage collected
         /// </summary>
-        public bool IsActive => !_isDisposed;
+        public bool IsActive => PinnedArray != null;
 
```

Comment:
I think IDisposable is no longer needed and PinnedArray can be a readonly field.

Reply:
Is needed to switch off the Slab; which then means its blocks are thrown away rather than returned to the pool (i.e. to take them out of circulation)

Reply:
Well sorta... since the pool doesn't really shrink; but if it did... 😉